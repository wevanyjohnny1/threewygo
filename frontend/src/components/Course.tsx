import { FaTrashAlt } from "react-icons/fa";
import { MdEdit } from "react-icons/md";

import { formatDate, sumLessonsTime } from "../shared/utils";
import {
  ActionContainer,
  Container,
  LeftContent,
  RightContent,
} from "../styles/components/course";

type Props = ICourse & {
  handleEditCourse: () => void;
  handleDeleteCourse: () => void;
};

const Course = ({
  endAt,
  lessons,
  name,
  handleEditCourse,
  handleDeleteCourse,
}: Props) => {
  return (
    <Container>
      <LeftContent>
        <strong>{name}</strong>
        <strong>{lessons.length} aulas</strong>
      </LeftContent>
      <RightContent>
        <strong>{sumLessonsTime(lessons)} minutos de aulas</strong>
        <strong>encerramento {formatDate(endAt)}</strong>
      </RightContent>

      <ActionContainer>
        <a data-testid="edit-course" onClick={handleEditCourse}>
          <MdEdit size={20} />
        </a>
        <a data-testid="delete-course" onClick={handleDeleteCourse}>
          <FaTrashAlt color="red" />
        </a>
      </ActionContainer>
    </Container>
  );
};

export default Course;
