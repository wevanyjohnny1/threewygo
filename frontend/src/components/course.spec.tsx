import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import Course from "./Course";

describe("Test course components", () => {
  const lessons = [
    {
      id: "1",
      duration: 30,
      title: "aula",
      url: "",
      courseId: "",
      createdAt: new Date(),
    },
    {
      id: "2",
      duration: 45,
      title: "aula 2",
      url: "",
      courseId: "",
      createdAt: new Date(),
    },
  ];

  const handleEditCourseMock = jest.fn();
  const handleDeleteCourseMock = jest.fn();

  function course() {
    return render(
      <Course
        id="1"
        name="Sample Course"
        description=""
        endAt="2024-01-31"
        image=""
        createdAt={new Date()}
        lessons={lessons}
        handleEditCourse={handleEditCourseMock}
        handleDeleteCourse={handleDeleteCourseMock}
      />
    );
  }

  it("should match with snapshot", () => {
    screen.debug();
    expect(course).toMatchSnapshot();
  });

  test("displays course details correctly", () => {
    const { getByText } = course();

    expect(getByText("Sample Course")).toBeInTheDocument();
    expect(getByText("2 aulas")).toBeInTheDocument();
    expect(getByText("75 minutos de aulas")).toBeInTheDocument();
    expect(getByText("encerramento 31/01/2024")).toBeInTheDocument();
  });
});
