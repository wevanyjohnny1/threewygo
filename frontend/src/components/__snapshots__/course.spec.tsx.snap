// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test course components should match with snapshot 1`] = `[Function]`;
